hist/hist/src/THnBase.cxx

```diff
@@ -282,6 +299,17 @@ TH1* THnBase::CreateHist(const char* name, const char* title,
             // uniform bins:
             hax[d]->Set(reqaxis->GetNbins(), reqaxis->GetXmin(), reqaxis->GetXmax());
          }
+         // Copy the axis labels if needed.
+         THashList* labels = reqaxis->GetLabels();
+         if (labels) {
+            TIter iL(labels);
+            TObjString* lb;
+            Int_t i = 1;
+            while ((lb=(TObjString*)iL())) {
+               hax[d]->SetBinLabel(i,lb->String().Data());
+               i++;
+            }
+         }
```

Comment:
Here it feels like the code could be simplified via something like (untested pseudo-code)
```cpp
if(auto *labels = reqaxis->GetLabels()){
int i = 0;
for (const auto *label: TRangeDynCast<TObjString>(labels){
    hax[d]->SetBinLabel(I, label->String().Data());
    i++;
}
}
```

And I'm feeling it could be simplified even further

Reply:
I get the error: `error: reference to type 'ROOT::Detail::TRangeCast<TObjString, true>' requires an initializer`

Reply:
Yeah, I didn't test the code, but it should be on me to provide the simpler implementation, yours works already :)